internal/app/apid/pkg/provider/tls.go

```diff
@@ -28,25 +30,39 @@ func NewTLSConfig(config runtime.Configurator, endpoints []string) (*TLSConfig,
 	if err != nil {
 		return nil, fmt.Errorf("failed to discover IP addresses: %w", err)
 	}
-	// TODO(andrewrynhard): Allow for DNS names.
+
+	domainName, err := ioutil.ReadFile("/proc/sys/kernel/domainname")
+	if err != nil {
+		return nil, err
+	}
+
+	dnsNames := []string{}
+
+	if strings.Contains(string(domainName), "(none)") {
+		var hostname string
+
+		if hostname, err = os.Hostname(); err != nil {
+			return nil, err
+		}
+
+		dnsNames = append(dnsNames, hostname, fmt.Sprintf("%s.%s", hostname, domainName))
+	}
+
```

Comment:
I think this should be in a helper func, but not quite sure _where_ to put it.

Reply:
It does kind of seem like something that might want to be reused, like say for the kubernetes side.  I'd classify it as network.

Reply:
Yeah, I mean I'm already using it twice in this PR.

Reply:
Not only that, but you might one day want to hook it up to a cloud provider's host metadata lookup for the hostname and/or IP.

Reply:
good point